sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java

```diff
@@ -197,6 +216,10 @@ public AutoConfiguredOpenTelemetrySdk build() {
             spanExporterCustomizer,
             samplerCustomizer);
 
+    if (registerShutdownHook) {
+      Runtime.getRuntime().addShutdownHook(new Thread(tracerProvider::close));
```

Comment:
`MeterProviderConfiguration` and `LogEmitterProviderConfiguration` also register shutdown hooks that should be conditional on `registerShutdownHook`.